LinqToDBPostGisNetTopologySuite.Tests/AffineTransformationsTests.cs

```diff
@@ -0,0 +1,258 @@
+                var result = db.TestGeometries
+                    .Where(g => g.Id == 1)
+                    .Select(g =>
+                        g.Geometry.STAffine(Math.Cos(Math.PI),
+                            -1 * Math.Sin(Math.PI), 0, Math.Sin(Math.PI), Math.Cos(Math.PI), 0, 0, 0, 1, 0, 0, 0)
+                    )
+                    .Select(g => GeometryOutput.STAsEWKT(g))
+                    .Single();
+                ;
+                Assert.AreEqual("LINESTRING(-1 -2 3,-1 -4 3)", result);
+            }
+        }
+
+        [Test]
+        public void TestSTAffine2D()
+        {
+            using (var db = new PostGisTestDataConnection(TestDatabaseConnectionString))
+            {
+                const string ewkt = "LINESTRING(1 2 3, 1 4 3)";
+                db.TestGeometries.Value(g => g.Id, 1).Value(p => p.Geometry, () => GeometryInput.STGeomFromEWKT(ewkt)).Insert();
+                var result = db.TestGeometries
+                    .Where(g => g.Id == 1)
+                    .Select(g =>
+                        g.Geometry.STAffine(0, 0, 0, 0, 0, 0)
+                    )
+                    .Select(g => GeometryOutput.STAsEWKT(g))
+                    .Single();
+                ;
+                Assert.AreEqual("LINESTRING(0 0 3,0 0 3)", result);
+            }
+        }
+        [Test]
+        public void TestSTRotate()
+        {
+            using (var db = new PostGisTestDataConnection(TestDatabaseConnectionString))
+            {
+                const string ewkt = "LINESTRING (50 160, 50 50, 100 50)";
+                db.TestGeometries.Value(g => g.Id, 1).Value(p => p.Geometry, () => GeometryInput.STGeomFromEWKT(ewkt)).Insert();
+                var result = db.TestGeometries
+                    .Where(g => g.Id == 1)
+                    .Select(g => g.Geometry.STRotate(Math.PI))
+                    .Select(g => GeometryOutput.STAsEWKT(g))
+                    .Single();
+                Assert.AreEqual(result, "LINESTRING(-50.00000000000002 -160,-50.00000000000001 -49.99999999999999,-100 -49.999999999999986)");
+            }
+        }
+        [Test]
+        public void TestSTRotateOrigin()
+        {
+            using (var db = new PostGisTestDataConnection(TestDatabaseConnectionString))
+            {
+                const string ewkt = "LINESTRING (50 160, 50 50, 100 50)";
+                db.TestGeometries.Value(g => g.Id, 1).Value(p => p.Geometry, () => GeometryInput.STGeomFromEWKT(ewkt)).Insert();
+                var result = db.TestGeometries
+                    .Where(g => g.Id == 1)
+                    .Select(g => g.Geometry.STRotate(Math.PI / 6.0, 50, 160))
+                    .Select(g => GeometryOutput.STAsEWKT(g))
+                    .Single();
+                Assert.AreEqual(result, "LINESTRING(50 160,104.99999999999999 64.73720558371174,148.30127018922192 89.73720558371173)");
```

Comment:
Test fails
Actual `result` is 
"LINESTRING(50 160,105 64.7372055837117,148.301270189222 89.7372055837117)"
Can the rounding depends on PostgreSQL/PostGIS version? 
I use PostgreSQL 9.6.1, PostGIS 3.0.2

Reply:
maybe.
i use postgis 3.1.0... potgresPostgreSQL 13.1 ; 
i will get my pg/postgis version sync to 9.6.1&3.0.2 and try fix the precision of geometry output

Reply:
Would be better to use `Assert.AreEqual` with precision tolerance on returned `Geometry` (not WKT/EWKT), instead of locking on specific versions.